### PR TITLE
IA-3482 IA-1694 IA-1728:  popups + remove Redux

### PR DESCRIPTION
**Pull request by beygorghor**
Orgunit popup and instance popup are acting weird while using clustering.

Those two component are the last one's to really use redux store. We can finally remove all redux related code

Related JIRA tickets : IA-3482   + IA-1694 + IA-1728

## Self proofreading checklist

- [ ] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- fix popups to use their own hook and fetch details infos
- custom hook to know if popup is open or not on map and trigger the call
- remove instances reducer, action
- remove org unit reducer and action
- remove all redux dependencies
- fix js test using redux

## How to test

Make sure you have two org units and two instances with the same location.
Open org unit search and display those two org units on the map, click on it, you should be able to open the popup with org units infos
Make the same thing with instances.

Play with org units list, detail, instances list, details

## Print screen / video

https://github.com/user-attachments/assets/90638959-9b20-4f1d-aa3c-76dfbbebadac

## Notes

Cypress tests are passing here : https://github.com/BLSQ/iaso/actions/runs/11405440785/job/31736948494

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/instances/actions.js
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/instances/actions.js:1]: YES!

File: hat/assets/js/apps/Iaso/domains/instances/components/ExportInstancesDialogComponent.js [hunk 1]
```diff
@@ -65,28 +59,9 @@ ExportInstancesDialogComponent.defaultProps = {
...
-            createExportRequest: createExportRequestAction,
-        },
-        dispatch,
-    ),
-});
-
-export default connect(
-    MapStateToProps,
-    MapDispatchToProps,
-)(injectIntl(ExportInstancesDialogComponent));
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/instances/components/ExportInstancesDialogComponent.js:92]: ❤️

File: hat/assets/js/apps/Iaso/domains/instances/index.js [hunk 2]
```diff
@@ -54,10 +53,10 @@ const Instances = () => {
     const params = useParamsObject(baseUrl);
     const classes = useStyles();
     const { formatMessage } = useSafeIntl();
-    const dispatch = useDispatch();
     const queryClient = useQueryClient();
     const redirectToReplace = useRedirectToReplace();
-
+    const [isInstancesFilterUpdated, setIsInstancesFilterUpdated] =
+        useState(false);
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/instances/index.js]: Couldn't we move this state to the child?

File: hat/assets/js/apps/Iaso/domains/instances/hooks/speedDialActions.tsx [hunk 3]
```diff
@@ -28,29 +28,24 @@ export type SpeedDialAction = {
...
                     <ExportInstancesDialogComponent
                         // @ts-ignore
-                        renderTrigger={(
-                            openDialog,
-                            isInstancesFilterUpdated,
-                        ) => (
+                        renderTrigger={openDialog => (
                             <ExportButton
                                 onClick={openDialog}
-                                isDisabled={isInstancesFilterUpdated}
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/instances/hooks/speedDialActions.tsx:53]: should we not disable it anymore?

**Review by quang-le — CHANGES_REQUESTED** [review]
Awesome job.! The architecture of these components is much less complicated now.

I just have a bug in Org Units > details> map tab: the org unit id is undefined so the pop up won't fetch and stays loading for ever:

https://github.com/user-attachments/assets/6dbb70c2-3876-452a-9c23-67316644793b

File: hat/assets/js/apps/Iaso/domains/instances/hooks/speedDialActions.tsx (hunk 3 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/instances/hooks/speedDialActions.tsx:53]: nope isInstancesFilterUpdated is for the list of submissions, this hook is used only on the detail page (SpeedDialInstance => details)

**Review by quang-le — APPROVED** [review]

**Pull request merged**